src/device-api/v1.ts

```diff
@@ -1,190 +1,213 @@
+					return applicationManager.getCurrentApps().then(function (apps2) {
+						const app2 = apps2[appId];
+						service = app2.services[0];
+						if (service == null) {
+							throw new Error('App not found after running action');
+						}
+						return service;
+					});
+				})
+				.then((service2) =>
+					res.status(200).json({ containerId: service2.containerId }),
+				);
+		})
+		.catch(next);
+};
+
+const createV1StopOrStartHandler = (action: 'start' | 'stop') =>
+	_.partial(v1StopOrStart, _, _, _, action);
+
+router.post('/v1/apps/:appId/stop', createV1StopOrStartHandler('stop'));
+router.post('/v1/apps/:appId/start', createV1StopOrStartHandler('start'));
+
+router.get('/v1/apps/:appId', async (req: AuthorizedRequest, res, next) => {
+	const appId = checkInt(req.params.appId);
+	eventTracker.track('GET app (v1)', { appId });
+	if (appId == null) {
+		return res.status(400).send('Missing app id');
+	}
+
+	try {
+		const apps = await applicationManager.getCurrentApps();
+		const app = apps[appId];
+		const service = app?.services?.[0];
+		if (service == null) {
+			return res.status(400).send('App not found');
 		}
 
 		// handle the case where the appId is out of scope
-		if (!req.auth.isScoped({ apps: [appId] })) {
+		if (!req.auth.isScoped({ apps: [app.appId] })) {
 			res.status(401).json({
 				status: 'failed',
 				message: 'Application is not available',
 			});
 			return;
 		}
 
-		return doPurge(appId, force)
-			.then(() => res.status(200).json({ Data: 'OK', Error: '' }))
+		if (app.services.length > 1) {
+			return res
+				.status(400)
+				.send('Some v1 endpoints are only allowed on single-container apps');
+		}
+
+		// Because we only have a single app, we can fetch the commit for that
+		// app, and maintain backwards compatability
+		const commit = await commitStore.getCommitForApp(appId);
+
+		// Don't return data that will be of no use to the user
+		const appToSend = {
+			appId,
+			commit,
+			containerId: service.containerId,
+			env: _.omit(service.config.environment, constants.privateAppEnvVars),
+			imageId: service.config.image,
+			releaseId: service.releaseId,
+		};
+
+		return res.json(appToSend);
+	} catch (e) {
+		next(e);
+	}
+});
+
+router.post('/v1/purge', (req: AuthorizedRequest, res, next) => {
+	const appId = checkInt(req.body.appId);
+	const force = checkTruthy(req.body.force);
+	if (appId == null) {
+		const errMsg = 'Invalid or missing appId';
+		return res.status(400).send(errMsg);
+	}
+
+	// handle the case where the appId is out of scope
+	if (!req.auth.isScoped({ apps: [appId] })) {
+		res.status(401).json({
+			status: 'failed',
+			message: 'Application is not available',
+		});
+		return;
+	}
+
+	return doPurge(appId, force)
+		.then(() => res.status(200).json({ Data: 'OK', Error: '' }))
+		.catch(next);
+});
+
+router.post('/v1/update', (req, res, next) => {
+	eventTracker.track('Update notification');
+	if (isReadyForUpdates()) {
```

Comment:
This is probably out of scope for this PR, but I am not sure this is necessary. Regardless of target state polling being started, the device API should be able to receive this request and try to trigger a target state request. If the device has not yet authenticated with the API, then the request will fail and the endpoint could return a 500

Reply:
Makes sense. I'll add this logic in a separate PR